### DISCUSSION TITLE
Error: Not found: 'dart:html'

### DISCUSSION THREAD

**Discussion by MarcusMendes81** [Flutterando/forum#110] [Perguntas e Respostas]
Olá pessoal, tudo bem? 

Toda vez quando vou tentar rodar a aplicação pelo simulador aparece este erro aqui:  *Error: Not found: 'dart:html'*

FileSystemExceptio
n(uri=org-dartlang-untranslatable-uri:dart%3Ahtml; message=StandardFileSystem only supports file:* a
nd data:* URIs)
#0      StandardFileSystem.entityForUri 
(package:front_end/src/api_
prototype/standard_file_system.dart:34:7)
#1      asFileUri (package:vm/kernel_front_end.dart:659:37)
#2      writeDepfile (package:vm/kernel_front_end.dart:799:21)
<asynchronous suspension>
#3      FrontendCompiler.compile (package:frontend_server/frontend_server.da

rt:625:9)
<asynchronous suspension>
#4      starter (package:frontend_server/frontend_server.dart:1451:12)
<asynchronous suspension>
#5      main (file:///C:/b/s/w/ir/x/w/sdk/pkg/frontend_server/bin/front
end_server_starter.dart:10:14)
<asynchronous suspension>
4

FAILURE: Build failed with an exception.

* Where:
Script 'C:\Users\maarc\Flutter\flutter\packages\flutter_tools\gradle\flutter.gradle' line: 1159

Eu já tentei fazer algumas coisas, como localizar esse import ou usar o comando flutter clean... Mas até agora nada.

Alguém tem ideia do que seja esse problema?

Uma observação pertinente (não sei se é)  é que quando vou depurar pelo browser, ele funciona, mas quando uso o simulador do android ou até mesmo meu próprio celular, dá esse erro aí.

Obrigado!

**Comment by Zeca-dev** [comment]
Esse problema já aconteceu comigo algumas vezes mas se tratava mesmo do import que em algum momento acabei colocando sem querer. Ao apagar o import resolveu. Uma dica é procurar na IDE, na aba erros, se tem os erros de imports não usados.

**Comment by CardosoGit** [comment]
Eu estava tendo o mesmo erro. Estava usando o código abaixo e precisava importar o dart:html, pois embora seja um webapp eu faço os testes no mobile :

```
@override
  void initState() {
    super.initState();
    WidgetsBinding.instance.addPostFrameCallback((_) => setState(() {
          if (isWeb) {
            if (MediaQuery.of(context).orientation == Orientation.landscape)
              document.documentElement?.requestFullscreen();
          }
        }));
    SystemChrome.setPreferredOrientations([
      DeviceOrientation.landscapeRight,
      DeviceOrientation.landscapeLeft,
    ]);
  }
```
  
  Resolvi adicionando o seguinte comentário:
 ```
 // ignore: avoid_web_libraries_in_flutter
import 'dart:html';
```

Obs. talvez seja necessário apagar a pasta 'build'.

Espero que ajude.

**Reply by CardosoGit** [reply]
> Eu estava tendo o mesmo erro. Estava usando o código abaixo e precisava importar o dart:html, pois embora seja um webapp eu faço os testes no mobile :
> 
> ```
> @override
>   void initState() {
>     super.initState();
>     WidgetsBinding.instance.addPostFrameCallback((_) => setState(() {
>           if (isWeb) {
>             if (MediaQuery.of(context).orientation == Orientation.landscape)
>               document.documentElement?.requestFullscreen();
>           }
>         }));
>     SystemChrome.setPreferredOrientations([
>       DeviceOrientation.landscapeRight,
>       DeviceOrientation.landscapeLeft,
>     ]);
>   }
> ```
> 
> Resolvi adicionando o seguinte comentário:
> 
> ```
> // ignore: avoid_web_libraries_in_flutter
> import 'dart:html';
> ```
> 
> Obs. talvez seja necessário apagar a pasta 'build'.
> 
> Espero que ajude.

Achei que tinha funcionado, mas infelizmente não deu certo. Ainda estou buscando uma solução.
